AWSAppSyncClient/AWSAppSyncCache.swift

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+// http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+import Foundation
+
+/// Errors thrown from the Queries Cache
+public enum AWSAppSyncQueriesCacheError: Error {
```

Comment:
Is there a way in which `AWSAppSyncQueriesCacheError` can inherit from `AWSSQLLiteNormalizedCacheError ` so that the customer experience can be slightly better? We can still deprecate `AWSSQLLiteNormalizedCacheError` and get rid of it eventually.

Reply:
I considered some options but couldn't find a way to make this a non-breaking change. Value types like Enums can't inherit in Swift, and any change to the throwing type is going to potentially impact customers who are catching that specifically.

The three options I considered:
1. Leave it as-is
2. Remove `AWSSQLLiteNormalizedCacheError` in favor of the new error type
3. Deprecate `AWSSQLLiteNormalizedCacheError` in favor of the new error type

I don't favor Option 1. `AWSSQLLiteNormalizedCacheError` as a type name has some semantic confusion (What's a normalized cache? Why is SQLLite specified? Why is SQLite misspelled? :) )

I went back & forth between Option 2 & Option 3 and only settled on deprecation because that's our traditional way of doing things. I could be convinced to do Option 2--just removing it altogether. Deprecating means that customers who are actually taking meaningful action based on the error type will see a warning, but won't be forced to deal with the potential change in behavior. However, for customers who aren't doing anything critical (e.g. they're adding or suppressing logging based on the error type), it's an unexpected change to remove it.

A deprecation with a FixIt seemed the best compromise.

Reply:
Sounds good, thanks for the detailed explanation!